src/main/java/uk/sky/cirrus/CqlMigratorImpl.java

```diff
@@ -0,0 +1,125 @@
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Standard implementation for {@code CqlMigrator}
+ *
+ * Implements Locking by providing mutex locking via the locks table.
+ * Client tries to acquire the lock and would only succeed for one thread.
+ * The inserted record will be deleted once the client releases the lock it holds, hence making way for a subsequent client thread to acquire the lock.
+ * Locking is done per client per keyspace. A client can obtain a lock on a different keyspace if no other client is holding a lock for that keyspace.
+ *
+ */
+
+public final class CqlMigratorImpl implements CqlMigrator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CqlMigratorImpl.class);
+
+    private final CassandraLockConfig lockConfig;
+
+    public CqlMigratorImpl(CassandraLockConfig lockConfig) {
+        this.lockConfig = lockConfig;
+    }
+
+    /**
+     * Allows cql migrate to be run on the command line.
+     * Set hosts, keyspace and directories using system properties.
+     */
+    public static void main(String[] args) {
+
+        final String hosts = System.getProperty("hosts");
+        final String keyspace = System.getProperty("keyspace");
+        final String directoriesProperty = System.getProperty("directories");
+        final String port = System.getProperty("port");
+
+        Preconditions.checkNotNull(hosts, "'hosts' property should be provided having value of a comma separated list of cassandra hosts");
+        Preconditions.checkNotNull(keyspace, "'keyspace' property should be provided having value of the cassandra keyspace");
+        Preconditions.checkNotNull(directoriesProperty, "'directories' property should be provided having value of the comma separated list of paths to cql files");
+
+        final Collection<Path> directories = Arrays.stream(directoriesProperty.split(","))
+                .map(Paths::get)
+                .collect(Collectors.toList());
+
+        new CqlMigratorImpl(CassandraLockConfig.builder().build())
+                .migrate(hosts.split(","), port == null ? 9042 : Integer.parseInt(port), keyspace, directories);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void migrate(String[] hosts, int port, String keyspace, Collection<Path> directories) {
+
+        try (Cluster cluster = createCluster(hosts, port);
+             Session session = cluster.connect()) {
+            this.migrate(session, keyspace, directories);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void migrate(Session session, String keyspace, Collection<Path> directories) {
+        final Cluster cluster = session.getCluster();
+        ClusterHealth clusterHealth = new ClusterHealth(cluster);
+        clusterHealth.check();
+
+        CassandraLockingMechanism cassandraLockingMechanism = new CassandraLockingMechanism(session, keyspace, lockConfig);
+        Lock lock = Lock.acquire(cassandraLockingMechanism, lockConfig);
```

Comment:
Do you think it's worth a comment why we don't do the lock in a `try {} finally {}` block? (I would usually expect an explicit lock in java to be like that).